File: mobile/library/objective-c/EnvoyHTTPFilterCallbacksImpl.m
#import "library/objective-c/EnvoyHTTPFilterCallbacksImpl.h"

#pragma mark - EnvoyHTTPFilterCallbacksImpl

@implementation EnvoyHTTPFilterCallbacksImpl {
  envoy_http_filter_callbacks _callbacks;
}

- (instancetype)initWithCallbacks:(envoy_http_filter_callbacks)callbacks {
  self = [super init];
  if (!self) {
    return nil;
  }

  _callbacks = callbacks;
  return self;
}

- (void)resumeIteration {
  _callbacks.resume_iteration(_callbacks.callback_context);
}

- (void)dealloc {
  _callbacks.release_callbacks(_callbacks.callback_context);
}

@end

File: mobile/library/swift/cxx/EnvoyHTTPFilterCallbacksImpl.swift
@_implementationOnly import EnvoyCxxSwiftInterop
@_implementationOnly import EnvoyEngine

final class EnvoyHTTPFilterCallbacksImpl: EnvoyHTTPFilterCallbacks {
  let callbacks: envoy_http_filter_callbacks
  init(callbacks: envoy_http_filter_callbacks) {
    self.callbacks = callbacks
  }

  func resumeIteration() {
    self.callbacks.resume_iteration(self.callbacks.callback_context)
  }

  func resetIdleTimer() {
    self.callbacks.reset_idle(self.callbacks.callback_context)
  }

  deinit {
    self.callbacks.release_callbacks(self.callbacks.callback_context)
  }
}
